plugin/hotswap-agent-spring-plugin/src/main/java/org/hotswap/agent/plugin/spring/BeanFactoryAssistant.java

```diff
@@ -0,0 +1,61 @@
+package org.hotswap.agent.plugin.spring;
+
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class BeanFactoryAssistant {
```

Comment:
It seems to me that 'BeanFactoryAssistant' is useless except for helping unit tests to find out how many reload times happens. I think we should remove this class, and find a better way to assert in unit test.

Reply:
> It seems to me that 'BeanFactoryAssistant' is useless except for helping unit tests to find out how many reload times happens. I think we should remove this class, and find a better way to assert in unit test.

My apology. I noticed there are two usages besides unit tests, one in ConfigurationClassPostProcessorTransformer, and the other in SpringChangedHub. 

I think we should still get rid of BeanFactoryAssistant since it makes more sense to let 'SpringReload' to hold 'isReload' status and 'placeHolderXmlRelation'.